src/protocol/sort/reveal.rs

```diff
@@ -0,0 +1,114 @@
+        prss::SpaceIndex,
+        Direction,
+    },
+    protocol::{context::ProtocolContext, RecordId, SortStep, Step},
+    secret_sharing::Replicated,
+};
+use embed_doc_image::embed_doc_image;
+use serde::{Deserialize, Serialize};
+
+/// A message sent by each helper when they've revealed their own shares
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct RevealValue<F> {
+    share: F,
+}
+
+/// This implements reveal algorithm
+/// For simplicity, we consider a simple revealing in which each Pi sends [a]i to Pi+1 and then reconstructs a from [a]i and
+/// [a]i−1.
+// Input: Each helpers know their own secret shares
+// Output: At the end of the protocol, all 3 helpers know a revealed (or opened) secret
+#[derive(Debug)]
+pub struct Reveal<F> {
+    input: Replicated<F>,
+}
+
+impl<F: Field> Reveal<F> {
+    #[allow(dead_code)]
+    pub fn new(input: Replicated<F>) -> Self {
+        Self { input }
+    }
+
+    /// Steps
+    /// Each helper sends their left share to the right helper. The helper then reconstructs their secret by adding the three shares
+    /// i.e. their own shares and received share.
+    #[embed_doc_image("reveal", "images/sort/reveal.png")]
+    #[allow(dead_code)]
+    pub async fn execute<M: Mesh, G: Gateway<M, S>, S: Step + SpaceIndex>(
+        self,
+        ctx: &ProtocolContext<'_, G, SortStep>,
+        record_id: RecordId,
+        step: S,
+    ) -> Result<F, BoxError> {
+        let mut channel = ctx.gateway.get_channel(step);
+
+        let inputs = self.input.as_tuple();
+        channel
+            .send(
+                channel.identity().peer(Direction::Right),
+                record_id,
+                RevealValue { share: inputs.0 },
+            )
+            .await?;
+
+        // Sleep until `to_helper.left` sends us their part2 value
+        let RevealValue { share } = channel
+            .receive(channel.identity().peer(Direction::Left), record_id)
+            .await?;
+
+        Ok(inputs.0 + inputs.1 + share)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use proptest::prelude::Rng;
+    use rand::rngs::mock::StepRng;
+    use tokio::try_join;
+
+    use crate::{
+        field::Fp31,
+        protocol::{sort::reveal::Reveal, QueryId, RecordId, SortStep},
+        test_fixture::{make_contexts, make_world, share, TestWorld},
+    };
+
+    #[tokio::test]
+    pub async fn reveal() {
+        let mut rand = StepRng::new(100, 1);
+
+        let mut rng = rand::thread_rng();
+
+        for _ in 0..10 {
+            let secret = rng.gen::<u128>();
+
+            let input = Fp31::from(secret);
+            let share = share(input, &mut rand);
+            let record_id = RecordId::from(1);
+
+            let world: TestWorld<SortStep> = make_world(QueryId);
+            let context = make_contexts(&world);
+
+            // TODO : change this step name to a test step
+            let step = SortStep::BitPermutations;
```

Comment:
@thurstonsand @akoshelev whats the best way to use a SortStep in a test environment? Ideally, would like to introduce a TestStep of type SortStep and then use it here. But SortStep is an enum and not a crate.